include/nbl/builtin/hlsl/bda/__ref.hlsl

```diff
@@ -0,0 +1,128 @@
+// Copyright (C) 2018-2024 - DevSH Graphics Programming Sp. z O.O.
+// This file is part of the "Nabla Engine".
+// For conditions of distribution and use, see copyright notice in nabla.h
+
+#include "nbl/builtin/hlsl/cpp_compat.hlsl"
+#include "nbl/builtin/hlsl/spirv_intrinsics/core.hlsl"
+
+#ifndef _NBL_BUILTIN_HLSL_BDA_REF_INCLUDED_
+#define _NBL_BUILTIN_HLSL_BDA_REF_INCLUDED_
+
+namespace spirv
+{
+template<typename M, typename T, typename StorageClass>
+[[vk::ext_instruction(/*spv::OpAccessChain*/65)]]
+vk::SpirvOpaqueType </* OpTypePointer*/ 32,StorageClass,M> accessChain(
+    [[vk::ext_reference]] vk::SpirvOpaqueType </* OpTypePointer*/ 32,
+    StorageClass,T>base,
+    [[vk::ext_literal]] uint32_t index0
+);
+}
+
+namespace bda
+{
+template<typename T>
+using __spv_ptr_t = vk::SpirvOpaqueType<
+    /* OpTypePointer */ 32,
+    /* PhysicalStorageBuffer */ vk::Literal<vk::integral_constant<uint,5349> >,
+    T
+>;
```

Comment:
btw when  you include `spirv_intrinsics/core.hlsl` you'll get access to spir-v headers, this means instead of doing silly things like `/* OpSomething */ literal` you can actually use the `spv::` enum, like all our other code

Reply:
Done